systemtest/src/test/java/io/strimzi/systemtest/operators/FeatureGatesST.java

```diff
@@ -0,0 +1,137 @@
+    public void testControlPlaneListenerFeatureGate(ExtensionContext extensionContext) throws InterruptedException {
+        final String clusterName = mapWithClusterNames.get(extensionContext.getDisplayName());
+        final String producerName = "producer-test-" + new Random().nextInt(Integer.MAX_VALUE);
+        final String consumerName = "consumer-test-" + new Random().nextInt(Integer.MAX_VALUE);
+        final String topicName = KafkaTopicUtils.generateRandomNameOfTopic();
+        int messageCount = 300;
+        List<EnvVar> testEnvVars = new ArrayList<>();
+        int kafkaReplicas = 3;
+
+        String systemEnvFG = System.getenv(Environment.STRIMZI_FEATURE_GATES_ENV);
+        if (systemEnvFG == null || !systemEnvFG.contains("ControlPlaneListener")) {
+            testEnvVars.add(new EnvVar(Environment.STRIMZI_FEATURE_GATES_ENV, "+ControlPlaneListener", null));
+        }
+        deployClusterOperatorWithEnvVars(extensionContext, testEnvVars);
+        resourceManager.createResource(extensionContext, KafkaTemplates.kafkaPersistent(clusterName, kafkaReplicas).build());
+
+        LOGGER.info("Check for presence of ContainerPort 9090/tcp (tcp-ctrlplane) in first Kafka pod.");
+        final Pod kafkaPod = PodUtils.getPodsByPrefixInNameWithDynamicWait(NAMESPACE, clusterName + "-kafka-").get(0);
+        ContainerPort expectedControlPlaneContainerPort = new ContainerPort(9090, null, null, "tcp-ctrlplane", "TCP");
+        List<ContainerPort> kafkaPodPorts = kafkaPod.getSpec().getContainers().get(0).getPorts();
+        assertTrue(kafkaPodPorts.contains(expectedControlPlaneContainerPort));
+
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(NAMESPACE, kafkaStatefulSetName(clusterName));
+
+        LOGGER.info("Try to send some messages to Kafka over the next {} seconds.", messageCount);
+        KafkaTopic kafkaTopic = KafkaTopicTemplates.topic(clusterName, topicName)
+                .editSpec()
+                    .withReplicas(kafkaReplicas)
+                    .withPartitions(kafkaReplicas)
+                .endSpec()
+                .build();
+        resourceManager.createResource(extensionContext, kafkaTopic);
+
+        KafkaBasicExampleClients kafkaBasicClientJob = new KafkaBasicExampleClients.Builder()
+                .withProducerName(producerName)
+                .withConsumerName(consumerName)
+                .withBootstrapAddress(KafkaResources.plainBootstrapAddress(clusterName))
+                .withTopicName(topicName)
+                .withMessageCount(messageCount)
+                .withDelayMs(500)
+                .withNamespaceName(NAMESPACE)
+                .build();
+
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi().build());
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.consumerStrimzi().build());
+        JobUtils.waitForJobRunning(consumerName, NAMESPACE, Duration.ofSeconds(20).toMillis());
```

Comment:
Just a nit - if you want to check that producer is running/sending messages, shouldn't it be 
```suggestion
        JobUtils.waitForJobRunning(producerName, NAMESPACE, Duration.ofSeconds(20).toMillis());
```
?

Reply:
Here I wanted to make sure (or rather assume), that if second started client is running, then first is running as well, so we can proceed with restarting broker pods. Fine by me with change or not.